source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,154 @@
+.. _golang-transactions:
+
+============
+Transactions
+============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use **transactions** with the
+{+driver-long+}. :manual:`Transactions </core/transactions/>` allow you
+to run a series of operations that are not visible from outside the
+transaction until it is committed. If any operation in the
+transaction fails, the driver aborts the transaction and discards all
+data changes before they ever become visible.
+
+In MongoDB, transactions run within logical **sessions**. A
+:manual:`session </reference/server-sessions/>` is a
+grouping of related read or write operations that you want to run
+sequentially. Sessions enable :manual:`causal consistency </core/read-isolation-consistency-recency/#causal-consistency>`
+for a group of operations or allow you to execute operations in an :website:`ACID
+transaction </basics/acid-transactions>`. MongoDB guarantees that the data
+involved in your transaction operations remains consistent, even if the operations
+encounter unexpected errors.
+
+With the {+driver-short+}, you can create a new session from a
+``Client`` instance as a ``Session`` type.
+
+.. warning::
+
+   You should only use ``Session`` with the ``Client`` (or associated
+   ``Database`` or ``Collection``) that created the ``Session``. Using a
+   ``Session`` with a different ``Client`` will result in operation
+   errors.
+
+.. warning::
+   
+   Implementations of ``Session`` are not safe for concurrent use by multiple `goroutines
+   <https://www.golang-book.com/books/intro/10>`__.
+
+Methods
+-------
+
+After you start a session using the ``StartSession()`` method, you can modify
+the session state using the method set provided by the ``Session`` interface. The
+following table describes these methods:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``StartTransaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Returns an error if there is already
+         a transaction in progress for the session. For more
+         information, see the :manual:`Manual entry </reference/method/Session.startTransaction/>`.
```

Comment:
Q: Does it make sense for "Manual" to be capitalized here? I'm not sure what we do in other docs.

Reply:
i see "Server Manual" capitalized elsewhere in the docs, but I also see manual entry with no capitalization. I will change these to lowercase as I am not including Server.